homeassistant/components/bluesound/media_player.py

```diff
@@ -447,9 +379,11 @@ async def async_update_status(self):
             raise
 
     @property
-    def unique_id(self):
+    def unique_id(self) -> str | None:
         """Return an unique ID."""
-        return f"{format_mac(self._sync_status['@mac'])}-{self.port}"
+        if self._sync_status is None:
+            return None
```

Comment:
When is sync status None?

Reply:
`SyncStatus` is one endpoint of the device. It starts with None and might be None if the device is not reachable.

Reply:
If the device is not reachable, I think we should raise an issue like PlatformNotReady I think

Reply:
I changed the `return None` to `raise PlatformNotReady`.

Reply:
No I mean, in the async_setup or async_setup_platform. If we can't fetch the state, that means the device doesn't respond and we can't get data. Right? In that case we should raise platform during setup

Reply:
Ah ok. I can do that. 

This integrations supports multiple devices. Throwing a `PlatformNotReady` would result in none of those working if one of the devices is not reachable.

Reply:
Right, let me ask some others how they did this back in the day

Reply:
Any updates on this?

And as some background information: Whatever we decide here will probably be removed in my other MR, which introduces a config flow and removes the yaml configuration.

Reply:
I did ask someone but got no reply, and I forgot to follow up as I was busy with my thesis, but let me ask someone else instead.